### ISSUE TITLE
[LINUX] Steam+Dota Crash upon clicking settings cog

### ISSUE THREAD

**Issue by yekyau** [ValveSoftware/Dota-2#1191]
Platform: Solus 2017.01.01
Problem: Clicking the settings cog in any menu or game state will cause a crash of both Steam and Dota.
* When the cog is clicked, the game will hang for up to a couple minutes before crashing. The game can still be killed from a terminal during the hang. Steam immediately crashes.
* The crash has persisted across:
  * Verifying dota files
  * Deleting and reinstalling dota
  * Clearing Steam download cache
  * Several reboots

The issue started yesterday to my knowledge, but I hadn't tried opening the settings for at least a couple days prior. I'm not sure how to get a proper logfile out of either dota or steam, but according to System Information I have uploaded 3 crash files.

System Information: https://gist.github.com/zJelly/108e630f8618781f41b1ed62961737a0

**Comment by kisak-valve** [comment]
Hello @zJelly, you can use the `-con_logfile` launch option to get Dota 2 to generate a logfile at game/dota/console.log.

Because the steam runtime is disabled on your system, you should also bring this issue to the attention of your package maintainer for steam. (Is that @ikeydoherty?)

**Comment by ikeydoherty** [comment]
@kisak-valve yep. This should've been raised through https://dev.solus-project.com/ to determine whether it was a Solus issue within the runtime or something for upstream.

In the mean time, @zJelly try disabling the Solus runtime using the LSI Settings shortcut in the menu, and then relaunch Steam completely (close from tray) - and try to reproduce the issue. This will then be using Valve's steam runtime.

**Comment by yekyau** [comment]
Hi all, thanks for the info. I've switched to the Steam runtime and repeated the crash twice with -con_logfile; once in the main menu and once in a hero demo. From what I could see in nautilus, the logfile's size did not change at all once the hang occured. I let it sit for a few minutes before the client closed itself.

New system information: https://gist.github.com/zJelly/ac3f28c28509a473d3d82e584196b3bb
(I switched to the steam client beta after the crashes had started, as a troubleshooting step)

Dota main menu crash: https://gist.github.com/zJelly/fd9c3375b1c481ba0e7dff212a49a5f5
Dota hero demo crash: https://gist.github.com/zJelly/e03961a97c4c535872b6e80be6f217a4

Other menus appear to work just fine, I can join arcade lobbies, load into games and play with a stable framerate without crashing.

**Comment by gdrewb-valve** [comment]
Things work fine on my Linux system.  The console output doesn't show anything interesting, but your crash reports listed in your system info are deep in libopenal and libpulse so this may be connected to your sound configuration.  That would be consistent with opening up settings since the sound settings will be queried.

**Comment by yekyau** [comment]
You seem to be onto something there! Performing a microphone test in the steam settings crashes my steam client (both with steam runtime and without). -nosound and -nomicsettings haven't prevented the crashes.

**Comment by xcava86x** [comment]
Hello,
I'm having a very similar issue on Manjaro, although I can test my microphone without issues in Steam.

When I'm playing Dota 2, if I try to open the option menu or push the key to talk in-game, both Dota 2 and Steam crash to desktop. I saved a log both running Steam through terminal and from Dota 2.

Perhaps is important to note that until a couple of weeks ago my microphone worked fine in Dota, and I haven't changed kernel or updated my system recently. Perhaps is important to note that I'm using a Sound Blaster X G5 soundcard?

System information: https://gist.github.com/xcava86x/0ddded2fd31c21bf9a6c4ddb1ae93639
Terminal output: https://gist.github.com/xcava86x/a798b97284b2107ea8236af7604131ae
Dota 2 log: https://gist.github.com/xcava86x/62c3f0d40eeb28ab9fdbee4bdbed454b

**Comment by kisak-valve** [comment]
Hello @xcava86x, you're most likely looking for https://github.com/ValveSoftware/steam-for-linux/issues/8930 instead of this issue report.
